lls/main.c

```diff
@@ -35,24 +56,69 @@ get_lls_conf(void)
 static int
 cleanup_lls(void)
 {
-	/* TODO Destroy ARP and ND cache. */
+	/* TODO Destroy ND cache. */
+	if (arp_enabled(&lls_conf))
+		lls_cache_destroy(&lls_conf.arp_cache);
 	destroy_mailbox(&lls_conf.requests);
 	destroy_mailbox(&lls_conf.updates);
 	rte_timer_stop(&lls_conf.timer);
 	return 0;
 }
 
+int
+resolve_arp(lls_update_cb cb, void *arg, struct in_addr *ip_be,
+	unsigned int lcore_id)
+{
+	if (arp_enabled(&lls_conf))
+		return lls_cache_request(&get_lls_conf()->arp_cache, cb, arg,
+			(uint8_t *)ip_be, lcore_id, LLS_REQUEST_RESOLVE);
+	else {
+		RTE_LOG(WARNING, GATEKEEPER, "lls: %s: lcore %u is requesting ARP service but ARP is not enabled\n",
+			__func__, lcore_id);
+		return -1;
+	}
+}
+
+int
+hold_arp(lls_update_cb cb, void *arg, struct in_addr *ip_be,
```

Comment:
This function is almost identical to the one before. I'd define a `static _hold_arp` that deals with the difference, namely, the last parameter of `lls_cache_request()`.